OpenSourcePolitics/decidim-mel
# [Backport] - FranceConnect Logout Modifications

### 🎩  Description

Backport of the modifications made on the decidim-app to fix correctly FranceConnect sessions management

Main reason:

- When we were disconnecting from our platform, our FranceConnect's session was still active and we needed to fix that.

### 📁 Modifications :

- Design modifications
    I. Modifications of buttons shared modal
    II. Modifications of the login modal to add FC button
    III. Optimizate displaying of images + text
    IV. Additions of few locales releated to FranceConnect
- Server-side modifications
    I. Modification of the config of omniauth_france_connect to make it work properly and manually specify the scope
    II. Extend Sessions and Account controllers to disconnect from FranceConnect when disconnecting from the platform
    III. Additions of few tests

### 🧪 How to test 

- Boot your platform
- Go to your system page
- Enable FranceConnect filling in `Client ID` `Client Secret` and `Site URL` and checking `enabled`
- Create an account using FranceConnect
- Connect to this account
- Disconnect or delete your account
- Check if you're still connected to FranceConnect

#### NB : If it works correctly, when you log out you should see a redirect link on your URL before returning to your platform which tells you that you are logged out of your platform.
